pkg/agent/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -538,6 +540,33 @@ func NewNetworkPolicyController(antreaClientGetter client.AntreaClientProvider,
 	return c, nil
 }
 
+func (c *Controller) GetFQDNCache(fqdnFilter *querier.FQDNCacheFilter) []types.DnsCacheEntry {
+	cacheEntryList := []types.DnsCacheEntry{}
+	var pattern *regexp.Regexp
+	var err error
+	if fqdnFilter != nil {
+		// have to convert human readable regex, i.e. *.example.com into regex that can be used
+		regexPattern := "^" + strings.ReplaceAll(regexp.QuoteMeta(fqdnFilter.Domain), `\*`, ".*") + "$"
+		pattern, err = regexp.Compile(regexPattern)
+		if err != nil {
+			// this pattern will match no strings if there is an error with the regex formatting or usage with the user specified --domain flag
+			pattern = regexp.MustCompile(`a\A`)
+		}
```

Comment:
we should handle such errors better, i.e., by returning a BadRequest HTTP code if the `domain` arg is not valid.

so maybe the regex conversion should happen in the handler and `FQDNCacheResponse` should include a regex instead of a string? cc @Dyanngg

Reply:
what about a warning displayed to the user but still an empty output?

Reply:
if it's an invalid pattern, it should be an error. A warning in the Antrea Agent logs is much harder to find. We should not treat 1) an unsupported pattern, and 2) a valid pattern that doesn't match anything, in the same way IMO.

Reply:
oops I thought you were done addressing comments, but looks like it is not the case
I will review again later